### DISCUSSION TITLE
javafx 文件选择器在V23版本不兼容

### DISCUSSION THREAD

**Discussion by lhDream** [linuxdeepin/developer-center#3837] [Troubleshooting | 故障排查] [answered]
在V23版本打开javafx的文件选器，文件选择器界面空白，且会导致系统UI卡死，并退出到系统登录界面，再次登录会黑屏，并且无法进入
demo
```java
public void start(Stage primaryStage) throws Exception{
    Button button = new Button();
    button.setOnAction(e->{
        FileChooser fileChooser = new FileChooser();
        fileChooser.showOpenDialog(primaryStage);
    });
    primaryStage.setScene(new Scene(button));
    primaryStage.show();
}
```
![image](https://user-images.githubusercontent.com/41568266/222032983-a9a069b9-9327-45ec-905c-10e5fc6011c3.png)

https://user-images.githubusercontent.com/41568266/222076125-1413502e-c2c0-4cbe-932d-b89891a39b88.mp4

**Comment by xzl01** [comment]
补充资料：https://docs.oracle.com/javafx/2/ui_controls/file-chooser.htm#:~:text=The%20FileChooser%20class%20is%20located%20in%20the%20javafx.stage,example%20of%20the%20file%20chooser%20dialog%20in%20Windows.

**Reply by lhDream** [reply]
https://github.com/openjdk/jfx

**Comment by chenchongbiao** [comment]
java 的版本以及jfx的版本是多少 我试试

**Reply by lhDream** [reply]
java8+，
我测试过的版本是 java8跟 java17两个版本，java17用的jfx版本是17.0.2

**Comment by lhDream** [comment]
可执行demo文件,java17,jfx_17.0.2
https://wwws.lanzouo.com/iBRoo0oxt02j
密码:bt09

**Reply by chenchongbiao** [reply]
你试过其他deepin或者uos上可以不 我得在v23配一下环境试试 我在uos上试jdk17 倒是可以

**Reply by chenchongbiao** [reply]
我试了 是你这个情况了

**Reply by lhDream** [reply]
其他版本正常喔...只有V23有这个问题

**Reply by chenchongbiao** [reply]
试了V23寄了 uos没事

**Reply by ComixHe** [reply]
能提供一下引起崩溃的最小复现代码吗？

**Reply by lhDream** [reply]
@ComixHe 我提供的demo就是全部的业务代码了

**Reply by lhDream** [reply]
核心就是这两行,用来启动文件选择器
```java
FileChooser fileChooser = new FileChooser();
fileChooser.showOpenDialog(primaryStage);
```

**Reply by ComixHe** [reply]
好的，我看一下

**Comment by Decodetalkers** [comment]
java调起文件管理器应该使用的也是`org.freedesktop.FileManager1`
因为在deepin上文件管理器和桌面环境是一个东西，所以这个请求把桌面炸掉了大概，所以不知道java这个请求发生了什么。我想也不是v23才出现的。

**Reply by lhDream** [reply]
这个不清楚，我测了专业版、跟家庭版没问题，目前就V23发现有这个问题

**Comment by chenchongbiao** [comment]
在v23上运行的时候就错误日志了，uos上试了没这个情况
```bash
(java:7091): GLib-GObject-CRITICAL **: 17:27:01.027: g_value_set_boxed: assertion 'G_VALUE_HOLDS_BOXED (value)' failed

(java:7091): dbind-WARNING **: 17:27:01.062: AT-SPI: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files

(java:7091): GLib-GObject-CRITICAL **: 17:27:01.140: g_value_set_boxed: assertion 'G_VALUE_HOLDS_BOXED (value)' failed
```

**Reply by lhDream** [reply]
这个报错是缺少一个库。。安装后不报错，但是还是会崩掉桌面

**Reply by lhDream** [reply]
我这里的报错是安装这个解决的 `sudo apt-get install gnome-themes-standard`

**Reply by xzl01** [reply]
你看看能否拿到文管的coredump信息 @chenchongbiao

**Reply by chenchongbiao** [reply]
我试试

**Reply by chenchongbiao** [reply]
> 这个报错是缺少一个库。。安装后不报错，但是还是会崩掉桌面

那就这不是主要问题 我用dbus监听下

**Comment by ComixHe** [comment]
崩溃的原因是调用dde-file-dialog后内存泄漏触发oom, 系统直接把很多进程都杀了，我今天想办法排查一下原因

**Reply by ComixHe** [reply]
bug为文件管理器相关实现问题，移交文管

**Comment by Decodetalkers** [comment]
可能解决了，你过几天试试

**Reply by lhDream** [reply]
好的，新的系统镜像已经发布了么

**Comment by xzl01** [comment, marked as answer]
https://bbs.deepin.org/post/256432  deepin v23已经修复此问题
